destination/config.go

```diff
@@ -33,4 +33,8 @@ type Config struct {
 	// record key string exceeds 256 it will be trimmed down from start to fit
 	// the partition key size
 	PartitionKeyTemplate string `json:"partitionKeyTemplate"`
+
+	// CreateIfNotExists makes the destination create a new stream if it does
+	// not exist when writing records.
+	CreateIfNotExists bool `json:"createIfNotExists" default:"true"`
```

Comment:
This one is fine, but when I've thought about this feature (auto creation), I was imagining having nested configuration to ultimately expand it for other operations and standardize across other connectors. 

Taking inspiration from [Debezium](https://debezium.io/documentation/reference/stable/configuration/topic-auto-create-config.html#disabling-automatic-topic-creation-for-the-kafka-broker
), I was thinking something like:

```go
AutoCreateStreamsEnable bool `json:"auto.create.streams.enable: default:"true"`
```

This way we could accommodate for other actions such as `auto.delete...` etc... and having the auto structure there.

cc @lovromazgon (in case you had also thought about this).

Reply:
I'm not sure about this one. How probable it is that we are going to have many more options like this one? Having a single option being 3 levels namespaced feels a bit too much.

Reply:
@alarbada I added [an item](https://www.notion.so/meroxa/Conduit-standup-7e4bb309856843d9b3018c9922cf6e33?pvs=4#40981118216c46f08f6d076b746e3aa3) to discuss in today's standup with the rest of the team.

Reply:
@alarbada after discussing in today's standup, we agreed that we should at least try to find something that could be consistent across the different connectors. Would something like `auto.create.streams` sound better to you? I feel like that's generic and flexible enough that could work for any other connector to ensure consistency.

Reply:
Yeah that could be it. I just get the 7 nested folders java vibes from it, which I despise, but I might be exaggerating.

Reply:
As a side note, auto create streams is a much better var name than what I came up, thanks :D